File: carrefourShop-backend/src/main/java/com/carrefour/carrefourShop/service/OrderService.java
package com.carrefour.carrefourShop.service;

import com.carrefour.carrefourShop.dto.CreateOrderRequest;
import com.carrefour.carrefourShop.dto.OrderDto;

import java.util.List;

public interface OrderService {
    OrderDto createOrder(Long userId, CreateOrderRequest request);
    List<OrderDto> getUserOrders(Long userId);
    OrderDto getOrderById(Long orderId, Long userId);
    OrderDto updateOrderStatus(Long orderId, String status);
}

File: carrfour-shop-frontend/src/services/orderService.ts
import api from './api';
import type { Order, CreateOrderData } from '../redux/types';

export const orderService = {
  createOrder: async (data: CreateOrderData) => {
    const response = await api.post<any>('/orders', data);
    return {
      ...response.data,
      totalAmount: response.data.totalPrice || response.data.totalAmount || 0,
    } as Order;
  },

  getUserOrders: async () => {
    const response = await api.get<any[]>('/orders');
    return response.data.map((order) => ({
      ...order,
      totalAmount: order.totalPrice || order.totalAmount || 0,
    })) as Order[];
  },

  getOrderById: async (id: number) => {
    const response = await api.get<any>(`/orders/${id}`);
    return {
      ...response.data,
      totalAmount: response.data.totalPrice || response.data.totalAmount || 0,
    } as Order;
  },

  updateOrderStatus: async (orderId: number, status: string) => {
    const response = await api.put<any>(`/orders/${orderId}/status?status=${status}`);
    return {
      ...response.data,
      totalAmount: response.data.totalPrice || response.data.totalAmount || 0,
    } as Order;
  },
};

